integration-tests/debezium-postgres/src/test/java/org/apache/camel/quarkus/component/debezium/postgres/it/DebeziumPostgresTestResource.java

```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.quarkus.component.debezium.postgres.it;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.camel.quarkus.testcontainers.ContainerResourceLifecycleManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+public class DebeziumPostgresTestResource implements ContainerResourceLifecycleManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebeziumPostgresTestResource.class);
+
+    private static final int POSTGRES_PORT = 5432;
+    private static final String POSTGRES_IMAGE = "debezium/postgres:11";
+
+    private PostgreSQLContainer<?> postgresContainer;
+    private Connection connection;
+    private Path storeFile;
+    private String hostname;
+    private int port;
+
+    @Override
+    public Map<String, String> start() {
+        LOGGER.info(TestcontainersConfiguration.getInstance().toString());
+
+        try {
+            storeFile = Files.createTempFile("debezium-postgress-store-", "");
+
+            postgresContainer = new PostgreSQLContainer<>(POSTGRES_IMAGE)
+                    .withUsername(DebeziumPostgresResource.DB_USERNAME)
+                    .withPassword(DebeziumPostgresResource.DB_PASSWORD)
+                    .withDatabaseName(DebeziumPostgresResource.DB_NAME)
+                    .withInitScript("init.sql");
+
+            postgresContainer.start();
+
+            final String jdbcUrl = "jdbc:postgresql://" + postgresContainer.getContainerIpAddress() + ":"
+                    + postgresContainer.getMappedPort(POSTGRES_PORT) + "/" + DebeziumPostgresResource.DB_NAME + "?user="
+                    + DebeziumPostgresResource.DB_USERNAME + "&password=" + DebeziumPostgresResource.DB_PASSWORD;
+            connection = DriverManager.getConnection(jdbcUrl);
+            hostname = postgresContainer.getContainerIpAddress();
+            port = postgresContainer.getMappedPort(POSTGRES_PORT);
+
+            return Collections.emptyMap();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void stop() {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+            if (postgresContainer != null) {
+                postgresContainer.stop();
+            }
+            if (storeFile != null) {
+                Files.deleteIfExists(storeFile);
+            }
+        } catch (Exception e) {
+            // ignored
+        }
+    }
+
+    @Override
+    public void inject(Object testInstance) {
+        ((DebeziumPostgresTest) testInstance).connection = this.connection;
```

Comment:
Do we need to do inject these things into the test instance? 

The usual way to set up properties needed by the component endpoint URIs would be to return a map of properties from `start()`. For example, like we do in [ActiveMQTestResource](https://github.com/apache/camel-quarkus/blob/master/integration-tests/messaging/src/test/java/org/apache/camel/quarkus/component/messaging/it/ActiveMQTestResource.java#L56-L63).

Reply:
@jamesnetherton  I know about Map approach. It is working well if property placeholder is "not" in test, where I could inject propertiesComponent to translate it for me. But I haven't found any way how to do it in test/* classes. Did I miss something?

Reply:
and for example connection is not a simple object

Reply:
I knew that it makes the map available to the application under test, but can the test code access it as well (how?).

Reply:
Aren't they just system properties? So you should be able to look them up.

Reply:
Indeed. I thought it was a bit more sophisticated. So assigning a random port to the app under test does not help at all to be able to run the tests in parallel, because the system props used to pass the port number are shared by all tests.

Reply:
@jamesnetherton I've started to work on debebezium-mysql. Because this extension will be very similar to debezium-postgres (and probably other debezium-* extensions). I'd like to refactor each of them to use some debezium-common extension. As a part of that refactor I'll cover also this issue with properties. -> From my POV I'd fixed it after merging this one PR.